### PR TITLE
feat(backend): Add qps & burst options in clientset. Fix #5095

### LINKED ISSUE #5095: Add kube-api-qps/kube-api-burst for pipeline components
In our internal clusters with KFP v1.0.0 release, we have over 20K workflow CRD in K8S Cluster. 

The pipeline API Server/Cache Server has significant performance downgrade because of K8S client-go default QPS/Burst configuration. We can see logs from client-go below (the default QPS/Burst is 5/10 if unspecified.):
```
Throttling request took xxxxx, request: xxxxxxxx
```

In such scenarios, I think we need add two more parameters (like kube-api-qps/kube-api-burst) for all backend components using client-go.

### PR DESCRIPTION
**Pull request by XXXXIEHF**
**Description of your changes:**
Fixes https://github.com/kubeflow/pipelines/issues/5095

Add flag options to specify QPS & burst of clients in multiple components including:
- API server
- cache server
- persistent agent
- scheduled workflow controller
- view controller

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by XXXXIEHF** [comment]
@googlebot I signed it!

File: backend/src/agent/persistence/main.go
Comment (Jeffwan) [inline backend/src/agent/persistence/main.go:137]: minor: use comment from client-go

"The maximum QPS to the master from this client."
"Maximum burst for throttle. from this client"

File: backend/src/apiserver/client_manager.go
Comment (Jeffwan) [inline backend/src/apiserver/client_manager.go:139]: Following initialization uses `common.GetXXX()` to retrieve configuration. Can you take a look if you can migrate to this consistent way?

**Review by Jeffwan — COMMENTED** [review]
Thanks for the contribution! Overall it looks good to me.

**Review by Jeffwan — APPROVED** [review]

File: backend/src/crd/controller/scheduledworkflow/main.go
Comment (NikeNano) [inline backend/src/crd/controller/scheduledworkflow/main.go:113]: Out of interest is there a reason for using `clientQPS` here and `client_qps` here: https://github.com/kubeflow/pipelines/blob/4373874925290230740508efaa47f1944528cc4c/backend/src/crd/controller/viewer/main.go#L49

File: backend/src/crd/controller/viewer/main.go
Comment (NikeNano) [inline backend/src/crd/controller/viewer/main.go:60]: Also a question to learn, it seems to be different between the different components how the flags are defined, `var` vs in `init` what is the reason for that?

File: backend/src/crd/controller/viewer/main.go
Comment (NikeNano) [inline backend/src/crd/controller/viewer/main.go:50]: For the default values, how did you decide on the values?

File: backend/src/crd/controller/scheduledworkflow/main.go
Comment (XXXXIEHF) [reply backend/src/crd/controller/scheduledworkflow/main.go:113]: Just try to keep consistent inside the component.

File: backend/src/crd/controller/viewer/main.go
Comment (XXXXIEHF) [reply backend/src/crd/controller/viewer/main.go:60]: Same as above, how the flags defined in different components is different, I try to keep them consistent inside a component.

File: backend/src/crd/controller/viewer/main.go
Comment (XXXXIEHF) [reply backend/src/crd/controller/viewer/main.go:50]: This is the default value set in client-go, there will be no difference if user does't set it manually.

File: backend/src/crd/controller/scheduledworkflow/main.go
Comment (NikeNano) [reply backend/src/crd/controller/scheduledworkflow/main.go:113]: Ok, thanks for the clarification.

File: backend/src/crd/controller/viewer/main.go
Comment (NikeNano) [reply backend/src/crd/controller/viewer/main.go:50]: Cool, thanks.

File: backend/src/agent/persistence/main.go
Comment (Bobgy) [inline backend/src/agent/persistence/main.go:137]: Would you mind adding a comment with link to default value of go-client?
I think that helps future maintainers to learn the context of how default is decided.

File: backend/src/apiserver/client/swf.go
Comment (Bobgy) [inline backend/src/apiserver/client/swf.go:44]: Is there an existing struct type we can use that includes clientQPS and clientBurst?
If not, shall we create a common struct type for our own use-case?

Using a struct as argument would be better for longer term maintenance, because it allows some arguments to be optional --- when we add more arguments, we don't need to change every test calling this function

**Review by Bobgy — COMMENTED** [review]
Thank you for opening the PR!

Did a quick review, and there are only some nit pickings.

/lgtm

/assign @capri-xiyue
to review too

File: backend/src/agent/persistence/main.go
Comment (capri-xiyue) [inline backend/src/agent/persistence/main.go:46]: Why do we use float64 here？ Looks like we use float32 for qps in other places？ Why don't we keep it consistent?

File: backend/src/apiserver/client/swf.go
Comment (capri-xiyue) [reply backend/src/apiserver/client/swf.go:44]: +1

File: backend/src/agent/persistence/main.go
Comment (capri-xiyue) [reply backend/src/agent/persistence/main.go:46]: NVM. There is no flag.Float32Var() method

**Comment by capri-xiyue** [comment]
/lgtm
(Just left some nits)

**Comment by capri-xiyue** [comment]
/lgtm
(Just left some nits)

**Comment by capri-xiyue** [comment]
/lgtm
(Just left some nits)

File: backend/src/agent/persistence/main.go
Comment (XXXXIEHF) [reply backend/src/agent/persistence/main.go:139]: Sure, added in new commit.

File: backend/src/apiserver/client/swf.go
Comment (XXXXIEHF) [reply backend/src/apiserver/client/swf.go]: Unfortunately there is not, I created a new struct named `util.ClientParameters` to contain clientQPS and clientBurst.

**Review by Bobgy — COMMENTED** [review]
/approve
Looks good to me.

/hold
I'll leave it to @capri-xiyue to make the final review, please unhold if you LGTM too.

**Pull request merged**
